src/encode.py

```diff
@@ -3,61 +3,73 @@
 
 from tqdm import tqdm
 
-FEATURES_DEFAULT = ('lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level')
 TIME_WINDOWS_DEFAULT = ('60s', '1h', '1d', '5d', '30d', '365d')
 COUNTERS_DEFAULT = ('attempts', 'wins')
 
-def encode_df(df: pd.DataFrame, features: tuple=FEATURES_DEFAULT, counters: tuple=COUNTERS_DEFAULT, time_windows: tuple=TIME_WINDOWS_DEFAULT)-> pd.DataFrame:
+
+def encode_df(df: pd.DataFrame, exercises: tuple = None, counters: tuple = COUNTERS_DEFAULT, time_windows: tuple = TIME_WINDOWS_DEFAULT) -> pd.DataFrame:
     '''
     Get the wanted counters of each student and dummifies the categorical variables of the dataset.
     '''
-    df = add_counters_for_all_features(df, features, counters, time_windows)
-    df = pd.get_dummies(df, columns=['student_id', 'lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level'])
+    if not exercises:
+        exercises = set(df['exercise_code'].unique())
+        # exercises done by only one student must be removed
```

Comment:
why?

Reply:
because the groupby is not working when there is only one student